.github/workflows/unit_tests.yaml

```
@@ -67,7 +67,7 @@ jobs:
     - name: Install Python packages
       run: |
           pip install --upgrade pip setuptools pytest pytest-xdist pytest-cov
-          pip install --upgrade flake8 "isort>=4.3.5" "mypy>=0.900" "click" "black"
+          pip install --upgrade .[dev]
```

We should instead do:
```suggestion
          pip install -r .github/workflows/style/requirements.txt
```
so that CI uses a stable version of all dependencies instead of random versions.

---

Yep, I think this is the better move too. `click` isn't present in `style/requirements.txt` however so it might be worth moving it to the prior line.

---

We should really do the same thing for the prior line to ensure that we never install the latest version of anything unless dependabot manually updates the version.

---

Done